### PR TITLE
Update sp-getapplock-transact-sql.md

### PR DESCRIPTION
**Pull request by 0x7FFFFFFFFFFFFFFF**
The original explanation contains some inaccuracies. It states, "This stronger lock mode is held until the lock is ultimately released even if lock release calls have occurred before that time." Let's break it down for clarity. The part "This stronger lock mode is held until the lock is ultimately released" is correct. In the original code, both "EXEC @result = sp_releaseapplock @Resource = 'Form1';" and "COMMIT TRANSACTION;" act to release the lock (if sp_releaseapplock has already released the lock, the commit operation won't affect the application lock since it no longer exists). However, the phrase "even if lock release calls have occurred before that time" seems to be misleading.

Consider this modification:

```
EXEC @result = sp_releaseapplock @Resource = 'Form1'; -- Newly added. Let's call it stmt1
EXEC @result = sp_releaseapplock @Resource = 'Form1';
COMMIT TRANSACTION;
```

In the phrase "before that time", "that time" refers to "the lock is ultimately released". If we insert stmt1 before the lock is ultimately released, would the lock still be held until it's finally released? No, it would be released by stmt1. Therefore, the original explanation is not entirely accurate. My suggested revision offers more clarity.

Moreover, manually releasing the application lock isn't necessary and isn't considered good practice. Observe the following code snippet:

```
EXEC @result = sp_releaseapplock @Resource = 'Form1';
-- <----- Other sessions can acquire the lock here, even though the transaction is still ongoing
COMMIT TRANSACTION;
```

Notice that once you've released the application lock with sp_releaseapplock, other sessions can acquire the lock, even while the transaction is still ongoing. However, the typical reason to use a transaction-level application lock is to maintain it for the duration of the entire transaction. There's no need to manually release it; the 'COMMIT TRANSACTION' statement will handle it for us. This revised version still communicates the same intent as the original example, but with greater precision and clarity.

### CODE REVIEW

**Comment by v-dirichards** [comment]
@markingmyname  Can you review this PR?

IMPORTANT: When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge on November 15th.



#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by rwestMSFT** [comment]
Removing the phrase is a good edit. Regarding the example code change, it might need some further thought (explanatory text), given the general misunderstanding of how transactions in the database engine work.

**Comment by 0x7FFFFFFFFFFFFFFF** [comment]
Maybe we can add a comment like this?
```
USE AdventureWorks2022;
GO
BEGIN TRANSACTION;
DECLARE @result int;
EXEC @result = sp_getapplock @Resource = 'Form1',
               @LockMode = 'Shared';
EXEC @result = sp_getapplock @Resource = 'Form1',
               @LockMode = 'Exclusive';
EXEC @result = sp_releaseapplock @Resource = 'Form1';
-- Since this is a transaction-level application lock, the COMMIT TRANSACTION statement below will automatically release it.
COMMIT TRANSACTION;
GO
```

**Comment by rwestMSFT** [comment]
We can work with that. I'll extract out the comment into the body of the article, so that it can be localized, but that's great.

I need to run this past a PM to confirm, and we're a little busy this time of year.

Thanks for your contribution!

**Comment by rwestMSFT** [comment]
After a review of the database engine source code and discussion with a PM, we cannot confirm that the code called by `sp_releaseapplock` is also executed by a `COMMIT`. I suspect this behaviour is coincidental, caused by a different code path.

However, your text change is valid, so I will modify this PR and then merge it based on that change.

File: docs/relational-databases/system-stored-procedures/sp-getapplock-transact-sql.md [hunk 1]
```diff
@@ -95,7 +95,6 @@ EXEC @result = sp_getapplock @Resource = 'Form1',
                @LockMode = 'Shared';  
 EXEC @result = sp_getapplock @Resource = 'Form1',   
                @LockMode = 'Exclusive';
```
Comment (rwestMSFT) [inline docs/relational-databases/system-stored-procedures/sp-getapplock-transact-sql.md]:
```suggestion
               @LockMode = 'Exclusive';  
EXEC @result = sp_releaseapplock @Resource = 'Form1';  
```

**Pull request merged**